vulnerability/worker.go

```diff
@@ -111,7 +113,7 @@ func (f *VulnerabilityWorker) Run(ctx context.Context) {
 			f.wg.Add(1)
 			go func() {
 				defer f.wg.Done()
-				err := f.fetcher.FetchInstances(ctx)
+				err := f.fetcher.FetchInstances(ctx, f.region)
```

Comment:
Instead of passing the region over and over again, a more clean solution will be to define the region once when instantiating the provider.
The usage of the multiRegion struct is good IMO as it's a more generic way to get data. We already have `MultiRegionClientFactory` that initializes the multi-region provider. Maybe we can add some methods to this struct that specify just certain regions instead of using all of them by default. For example:
```
WithRegions(region ...string)
WithDefaultRegion()
```
What do you think?